app/helpers/course/controller_helper.rb

```diff
@@ -27,4 +27,23 @@ def link_to_course_user(user, options = {})
       end
     end
   end
+
+  # Links to the given User or CourseUser. If a User is given, the CourseUser under
+  # current_course of the given user will be displayed.
+  #
+  # @param [CourseUser|User] user The User to display.
+  # @param [Hash] options The options to pass to +link_to+
+  # @param [Proc] block The block to use for displaying the user.
+  # @return [String] The user-visible string, including embedded HTML which will display the
+  #   string within a link to bring to the User page.
+  def link_to_user(user, options = {}, &block)
```

Comment:
Does this make sense to you ? Basically I think we need to call `super` and link to the User when course_user cannot be found. (i.e. when admin creates something, there is no course_user)